### PR TITLE
[ML-5994] Update Pregel API docs (Scala)

### PR DESCRIPTION
**Pull request by mengxr**
* moved `Pregel` to `graphframes.lib`.
* updated all API doc in Scala
* added example code to API doc
* No real code changes.

After this one gets reviewed and merged, I will send a follow-up PR for Python API doc changes.

### CODE REVIEW

File: src/test/scala/org/graphframes/lib/PregelSuite.scala [hunk 1]
```diff
@@ -46,7 +48,7 @@ class PregelSuite extends SparkFunSuite with GraphFrameTestSparkContext {
       .withVertexColumn("rank", lit(1.0 / numVertices),
         coalesce(Pregel.msg, lit(0.0)) * (1.0 - alpha) + alpha / numVertices)
       .sendMsgToDst(Pregel.src("rank") / Pregel.src("outDegree"))
-      .aggMsgs(sum(Pregel.MSG_COL_NAME))
+      .aggMsgs(sum(Pregel.msg))
```
Comment (mengxr) [inline src/test/scala/org/graphframes/lib/PregelSuite.scala]: This is changed to verify the example code is correct.

File: src/main/scala/org/graphframes/lib/Pregel.scala [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+ * processing</a> for a detailed description of the Pregel algorithm.
+ *
+ * You can construct a Pregel instance using either this constructor or [[org.graphframes.GraphFrame#pregel]],
+ * then use builder pattern to describe the operations, and then call [[run]] to start a run.
+ * It returns a DataFrame of vertices from the last iteration.
+ *
+ * When a run starts, it expands the vertices DataFrame using column expressions defined by [[withVertexColumn]].
+ * Those additional vertex properties can be changed during Pregel iterations.
+ * In each Pregel iteration, there are three phases:
+ *   - Given each triplet, generate messages and targets to send,
```
Comment (WeichenXu123) [inline src/main/scala/org/graphframes/lib/Pregel.scala]: `targets` => `corresponding target vertex IDs` ? .

File: src/main/scala/org/graphframes/lib/Pregel.scala [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+  private val CHECKPOINT_NAME_PREFIX = "pregel"
+
+  /** Sets the max number of iterations (default: 10). */
+  def setMaxIter(value: Int): this.type = {
+    maxIter = value
+    this
+  }
+
+  /**
+   * Sets the number of iterations between two checkpoints (default: 2).
```
Comment (WeichenXu123) [inline src/main/scala/org/graphframes/lib/Pregel.scala]: What about change default value to be `1` ? (In my test `localCheckpoint` will be fast and this change will probably save SQL planner time, especially when there're many partitions in dataset.)

File: src/main/scala/org/graphframes/lib/Pregel.scala [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+ * You can construct a Pregel instance using either this constructor or [[org.graphframes.GraphFrame#pregel]],
+ * then use builder pattern to describe the operations, and then call [[run]] to start a run.
+ * It returns a DataFrame of vertices from the last iteration.
+ *
+ * When a run starts, it expands the vertices DataFrame using column expressions defined by [[withVertexColumn]].
+ * Those additional vertex properties can be changed during Pregel iterations.
+ * In each Pregel iteration, there are three phases:
+ *   - Given each triplet, generate messages and targets to send,
+ *     described by [[sendMsgToDst]] and [[sendMsgToSrc]].
+ *   - Aggregate messages by destination vertex IDs, described by [[aggMsgs]].
```
Comment (WeichenXu123) [inline src/main/scala/org/graphframes/lib/Pregel.scala]: `destination vertex IDs` => `target vertex IDs` ?

**Pull request merged**